### PR TITLE
AC_WPNav: Stop parameter changes overriding speeds

### PR DESCRIPTION
**Pull request by lthall**
When we made parameter changed act automatically we failed to appreciate that WP_Nav is used by a number of different flight modes that do not use WP_Nav speeds. This means that changes in these parameters at any time will override the settings for RTL for example.

### CODE REVIEW

**Comment by rmackay9** [comment]
It looks like we have some auto tests that specifically test that changing the parameters takes effect immediately so these will need to be removed.

FYI @peterbarker

**Comment by IamPete1** [comment]
Seems a shame to loose this functionality in all auto. Could we move it to a new function to only be called in auto?

**Comment by rmackay9** [comment]
Discussed on the dev call and I will create a new PR which avoids the bug and then raise an issue for QGC and MP to add or improve the GCS's do-change-speed support and then we can merge this PR (or something like it) to remove the user's ability to change the vehicle's speed by modifying the WPNAV_SPEED parameter

File: Tools/autotest/arducopter.py [hunk 1]
```diff
@@ -3832,86 +3832,6 @@ def SplineTerrain(self):
         self.set_parameter("TERRAIN_ENABLE", 0)
         self.fly_mission("wp.txt")
 
-    def WPNAV_SPEED(self):
```
Comment (IamPete1) [inline Tools/autotest/arducopter.py:3835]: Also needs removing here: https://github.com/ArduPilot/ardupilot/blob/56773f09b422ece063b3531213e32618d04b1319/Tools/autotest/arducopter.py#L11740

Comment (IamPete1) [reply Tools/autotest/arducopter.py:3835]: Also note that we have tests for `WPNAV_SPEED_UP` and `WPNAV_SPEED_DN`, do they need updating?

**Comment by tridge** [comment]
should we have a separate RTL_SPEED, in m/s, which if 0 uses WPNAV_SPEED ?

**Pull request closed without merging**